bin/pkg/validate_lock_dir.ml

```diff
@@ -0,0 +1,79 @@
+open! Import
+open Pkg_common
+module Package_universe = Dune_pkg.Package_universe
+module Lock_dir = Dune_pkg.Lock_dir
+module Opam_repo = Dune_pkg.Opam_repo
+module Package_version = Dune_pkg.Package_version
+module Opam_solver = Dune_pkg.Opam_solver
+
+let info =
+  let doc = "Validate that a lockdir contains a solution for local packages" in
+  let man = [ `S "DESCRIPTION"; `P doc ] in
+  Cmd.info "validate-lockdir" ~doc ~man
+;;
+
+let enumerate_lock_dirs_by_path ~context_name_arg ~all_contexts_arg =
+  let open Fiber.O in
+  let+ per_contexts =
+    Per_context.choose ~context_name_arg ~all_contexts_arg ~version_preference_arg:None
+  in
+  List.filter_map per_contexts ~f:(fun { Per_context.lock_dir_path; _ } ->
+    if Path.exists (Path.source lock_dir_path)
+    then (
+      try Some (lock_dir_path, Lock_dir.read_disk lock_dir_path) with
+      | User_error.E e ->
+        User_warning.emit
+          [ Pp.textf
+              "Failed to parse lockdir %s:"
+              (Path.Source.to_string_maybe_quoted lock_dir_path)
+          ; User_message.pp e
+          ];
+        None)
+    else None)
+;;
+
+let validate_lock_dirs ~context_name_arg ~all_contexts_arg =
+  let open Fiber.O in
+  let+ lock_dirs_by_path = enumerate_lock_dirs_by_path ~context_name_arg ~all_contexts_arg
+  and+ local_packages = find_local_packages in
+  if List.is_empty lock_dirs_by_path
+  then User_error.raise [ Pp.text "No lockdirs to validate." ];
```

Comment:
It's a good question what the result should be if there is no lock dirs. Is the absence of any lock dirs an error or is it a success because all 0 lock dirs are valid?

I lean towards the latter but maybe one wouldn't call `validate-lock-dir` if one did not expect lock dirs to exist. OTOH, having to enable or disable validation explicitly in e.g. scripts is annoying.

Reply:
Good point about scripts. I'll update it to print a message but exit with success.